extension/auth/client.go

```diff
@@ -20,9 +20,21 @@ import (
 	"google.golang.org/grpc/credentials"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
 )
 
-var _ Client = (*defaultClient)(nil)
```

Comment:
Where did this go?

Reply:
Not needed (in my opinion) because this is compile time check that defaultClient implements Client, but the same check (compe time) is in NewClient since we return defaultClient and return type is Client